api/rpc/project/project.proto

```diff
@@ -2,15 +2,30 @@ syntax = "proto3";
 
 package project;
 
-service Project {
-  rpc Create (CreateRequest) returns (CreateReply) {}
+service ProjectSvc {
+  rpc Create (Project) returns (ProjectReply) {}
+  rpc Delete (Project) returns (ProjectReply) {}
+  rpc List   (Empty) returns (ProjectsReply) {}
+  rpc Get   (Key) returns  (ProjectReply) {}
+  rpc Update (Project) returns (ProjectReply) {}
 }
 
-message CreateRequest {
-  string id = 1;
-  string name = 2;
+message Project {
+  int32  repo_id = 1;
+  string owner_name = 2;
+  string repo_name = 3;
+  string token = 4;
 }
+message Key {
+  int32 repo_id=1;
+}
+message Empty{}
 
-message CreateReply {
-  string message = 1;
+message ProjectReply {
+    Project project=1;
+    string error_msg=2;
```

Comment:
I think errors should not be included in messages since gRPC provides an error mechanism for each RPC calls already.